Source/JavaScriptCore/runtime/JSStringInlines.h

```diff
@@ -177,28 +180,87 @@ void JSRopeString::resolveToBufferSlow(Fibers* fibers, CharacterType* buffer, un
 }
 
 template<typename Fibers, typename CharacterType>
-inline void JSRopeString::resolveToBuffer(Fibers* fibers, CharacterType* buffer, unsigned length)
+inline void JSRopeString::resolveToBuffer(Fibers* fibers, CharacterType* buffer, unsigned length, uint8_t* stackLimit)
 {
-    for (size_t i = 0; i < JSRopeString::s_maxInternalRopeLength; ++i) {
-        JSString* fiber = fibers->fiber(i);
-        if (!fiber)
-            break;
-        if (fiber->isRope()) {
-            JSRopeString::resolveToBufferSlow(fibers, buffer, length);
-            return;
+    if (UNLIKELY(bitwise_cast<uint8_t*>(currentStackPointer()) < stackLimit))
+        return JSRopeString::resolveToBufferSlow(fibers, buffer, length);
+
+    CharacterType* position = buffer;
```

Comment:
Does it make sense to add a `static_assert(3 == JSRopeString::s_maxInternalRopeLength)` here to flag if we change rope length.

Reply:
Nice, added.